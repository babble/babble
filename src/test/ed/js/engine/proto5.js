
A = function(){
    this.thingy = 2;
};
A.prototype.junk = 45;
A.prototype.word = 4;

a = new A();
print(a.junk);
print(a.thingy);
print(a.word);
print( a.prototype ? "yes" : "no" );

print( "---" );

B = function(){
    this.thingy = 1;
    this.zzz = "zzz";
};

B.prototype = new A();
//B.prototype.constructor = B;
B.prototype.junk = 54;

b = new B();
print(b.junk);
print(b.thingy);
print(b.word);
print(b.zzz);

print( "---" );

a = new A();
print(a.junk);
print(a.thingy);
print(a.word);
