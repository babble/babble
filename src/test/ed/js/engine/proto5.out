45
2
4
no
---
54
1
4
zzz
---
45
2
4
